google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/CredentialsService.java

```diff
@@ -20,23 +20,17 @@
 import com.google.cloud.spanner.ErrorCode;
 import com.google.cloud.spanner.SpannerException;
 import com.google.cloud.spanner.SpannerExceptionFactory;
-import com.google.cloud.storage.Blob;
-import com.google.cloud.storage.Storage;
-import com.google.cloud.storage.StorageOptions;
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Preconditions;
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.IOException;
 import java.io.InputStream;
-import java.nio.channels.Channels;
 
-/** Service class for getting credentials from key files stored locally or on cloud storage. */
+/** Service class for getting credentials from key files. */
 class CredentialsService {
-  private static final String GOOGLE_CLOUD_STORAGE_PREFIX = "gs://";
-  private static final String INVALID_GCS_PREFIX_MSG =
-      String.format("Storage URL must start with %s", GOOGLE_CLOUD_STORAGE_PREFIX);
-
+  static final String GCS_NOT_SUPPORTED_MSG =
+      "Credentials that are stored on Google Cloud Storage is no longer supported. Please use the Google Cloud Storage client library to download the credentials to a local file and reference the local file in the connection URL.";
```

Comment:
is --> are
No "Please"

http://www.textdoctor.com/webinars/please/

Most important, I don't think it's the google cloud storage client library that will be used to download these credentials. Perhaps just "download the credentials to a local file and reference the local file in the connection URL."

Reply:
Fixed.